servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/NettyHttpServer.java

```diff
@@ -649,7 +663,33 @@ private static void logDecoderException(final DecoderException e,
             }
         }
 
-        private static void logUnexpectedException(final Throwable t, NettyHttpServerConnection connection) {
+        private static void logHttp2Exception(final Http2Exception e, final NettyHttpServerConnection connection) {
+            final Http2ErrorCode errorCode = e.errorCode();
+            if (errorCode == CANCEL) {
+                LOGGER.debug(
+                        "{} HTTP/2 stream was cancelled by a remote peer, most likely due to timeout or lost interest",
+                        connection, e);
+            } else if (errorCode == NO_ERROR) {
+                LOGGER.debug("{} HTTP/2 stream was closed because underlying connection closed due to GO_AWAY",
+                        connection, e);
+            } else if (errorCode == SETTINGS_TIMEOUT) {
+                LOGGER.warn("{} HTTP/2 stream was closed because underlying connection did not receive SETTINGS " +
+                        "acknowledgement on time", connection, e);
+            } else if (errorCode == PROTOCOL_ERROR || errorCode == INTERNAL_ERROR || errorCode == FLOW_CONTROL_ERROR ||
+                    errorCode == STREAM_CLOSED || errorCode == FRAME_SIZE_ERROR || errorCode == COMPRESSION_ERROR ||
```

Comment:
Are we expecting to receive RST_STREAM(STREAM_CLOSED) due to half closed races? Eg, maybe the client sends a RST_STREAM(CANCEL) frame which races with some DATA frames we already sent, and those materialize into RST_STREAM(STREAM_CLOSED)?

I recall seeing that condition pretty commonly in the past but I don't know if it will surface here since we may have dropped the stream already due to the initial CANCEL.